src/vellum/lib/test_suites/resources.py

```diff
@@ -0,0 +1,134 @@
+
+        all_metric_results = [execution.metric_results for execution in executions]
+
+        filtered_metric_outputs = [
+            metric_result.outputs
+            for metric_result_execution in all_metric_results
+            for metric_result in metric_result_execution
+            if not metric_identifier
+            or (metric_result.metric_id == metric_identifier)
+            or (metric_result.metric_label == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.id == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.name == metric_identifier)
+            or (metric_result.metric_definition and metric_result.metric_definition.label == metric_identifier)
+        ]
+
+        filtered_metric_filtered_outputs = [
+            metric_output
+            for metric_output_execution in filtered_metric_outputs
+            for metric_output in metric_output_execution
+            if not output_identifier or metric_output.name == output_identifier
+        ]
+
+        return filtered_metric_filtered_outputs
+
+    def _refresh_test_suite_run(self):
+        test_suite_run = self.client.test_suite_runs.retrieve(self.id)
+        self._state = test_suite_run.state
+
+    def _get_test_suite_run_executions(self):
+        if self._executions is not None:
+            return self._executions
+
+        start_time = time.time_ns()
+        while True:
+            self._refresh_test_suite_run()
+            if self._state not in {TestSuiteRunState.QUEUED, TestSuiteRunState.RUNNING}:
+                break
+
+            current_time = time.time_ns()
+            if ((current_time - start_time) / 1e6) > self._max_polling_duration:
+                raise TestSuiteRunResultsException("Test suite run timed out polling for executions")
+
+            time.sleep(self._polling_interval / 1000.0)
+
+        if self._state == TestSuiteRunState.FAILED:
+            raise TestSuiteRunResultsException("Test suite run failed")
+
+        if self._state == TestSuiteRunState.CANCELLED:
+            raise TestSuiteRunResultsException("Test suite run was cancelled")
+
+        response = self.client.test_suite_runs.list_executions(
+            self.id, expand=["results.metric_results.metric_definition", "results.metric_results.metric_label"]
+        )
+        self._executions = response.results
+        return self._executions
+
+
+class VellumTestSuite:
+    def __init__(
+        self,
+        id: str | None = None,
+        # TODO: Support APIs that query by name
+        # https://app.shortcut.com/vellum/story/2689
+        # name: str | None = None,
+    ) -> None:
+        self.client = Vellum(
+            api_key=os.environ.get("VELLUM_API_KEY"),
+        )
+        self._id = id
+
+    def run_external(
+        self, executable: Callable[[list[TestCaseVariableValue]], list[NamedTestCaseVariableValueRequest]]
```

Comment:
I think this is the right call signature for this initial PR, but I would love to _also_ support a call signature that uses primitive python types and under the hood, use a common util function we define somewhere to "serialize" / "deserialize" into our strict types.

Reply:
Yea I have ideas there